File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/contactnotifier/RemoteNotificationRequest.java
package org.elastos.trinity.runtime.contactnotifier;

import org.json.JSONException;
import org.json.JSONObject;

public class RemoteNotificationRequest {
    /** Identification key used to overwrite a previous notification if it has the same key. */
    public String key = null;
    /** Package ID of the sending app. */
    public String appId = null;
    /** Title to be displayed as the main message on the notification. */
    public String title = null;
    /** Intent URL emitted when the notification is clicked. */
    public String url = null;

    public static RemoteNotificationRequest fromJSONObject(JSONObject obj) {
        try {
            RemoteNotificationRequest notif = new RemoteNotificationRequest();
            if (obj.has("key"))
                notif.key = obj.getString("key");
            if (obj.has("appId"))
                notif.appId = obj.getString("appId");
            if (obj.has("title"))
                notif.title = obj.getString("title");
            if (obj.has("url"))
                notif.url = obj.getString("url");
            return notif;
        }
        catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: Trinity.Runtime/platform_src/ios/elastOS/Classes/ContactNotifier/RemoteNotificationRequest.swift

public class RemoteNotificationRequest {
    /** Identification key used to overwrite a previous notification if it has the same key. */
    public var key: String? = nil
    /** Package ID of the sending app. */
    public var appId: String? = nil
    /** Title to be displayed as the main message on the notification. */
    public var title: String? = nil
    /** Intent URL emitted when the notification is clicked. */
    public var url: String? = nil

    public static func fromJSONObject(_ obj: Dictionary<String, Any>) -> RemoteNotificationRequest? {
        guard obj.keys.contains("key") && obj.keys.contains("title") else {
            return nil
        }
        
        let notif = RemoteNotificationRequest()
        
        notif.key = obj["key"] as? String
        
        if obj.keys.contains("appId") {
            notif.appId = obj["appId"] as? String
        }
        
        notif.title = obj["title"] as? String
        
        if obj.keys.contains("url") {
            notif.url = obj["url"] as? String
        }
        
        return notif
    }
}
